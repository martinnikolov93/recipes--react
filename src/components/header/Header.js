import React, { useContext } from 'react'

import { NavLink } from 'react-router-dom'

import styles from './Header.module.css'
import UserContext from '../../Context'
import CategoriesDropdown from '../categories-dropdown/CategoriesDropdown'

const Header = () => {
    const context = useContext(UserContext)

    return (
        <header>

            <h2 className="app-title"><img className="logo" src="/logo48.png" alt="logo" />React Recipes</h2>

            <nav>
                <div className={styles['nav-buttons-wrapper']}>
                    <NavLink className={styles['nav-button']} to='/' exact activeClassName={styles['nav-button-active']}>Home</NavLink>
                    <CategoriesDropdown className={styles['nav-button']} />
                    {context.loggedIn
                        ?
                        <>
                            <NavLink className={styles['nav-button']} to='/recipe/add' activeClassName={styles['nav-button-active']}>Add Recipe</NavLink>
                            <NavLink className={styles['nav-button']} to='/favourites' activeClassName={styles['nav-button-active']}>Favourites</NavLink>
                            <NavLink className={styles['nav-button']} to='/profile' activeClassName={styles['nav-button-active']}>Profile</NavLink>
                            <NavLink className={styles['nav-button']} to='/logout'>Logout</NavLink>
                        </>
                        :
                        <>
                            <NavLink className={styles['nav-button']} to='/login' activeClassName={styles['nav-button-active']}>Login</NavLink>
                            <NavLink className={styles['nav-button']} to='/register' activeClassName={styles['nav-button-active']}>Register</NavLink>
                        </>
                    }
                </div>
            </nav>
        </header>

    )
}

export default Header