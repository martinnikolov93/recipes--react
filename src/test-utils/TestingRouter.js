import React from 'react';
import UserContext from '../Context';
import { BrowserRouter, Route } from 'react-router-dom'

const TestingRouter = ({user, children}) => {
    return (
        <BrowserRouter>
            <Route>
                <UserContext.Provider value={user}>
                    {children}
                </UserContext.Provider>
            </Route>
        </BrowserRouter>
    )
}

export default TestingRouter