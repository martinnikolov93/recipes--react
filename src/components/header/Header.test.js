import React from 'react';
import renderer from 'react-test-renderer'
import Header from './Header';
import TestingRouter from '../../test-utils/TestingRouter';

const userAuthDummy = {
    loggedIn: true
}

const userUnauthDummy = {
    loggedIn: false
}

describe('Header component', () => {
    it('Should render authenticated header component', () => {
        const tree = renderer.create(
            <TestingRouter user={userAuthDummy}>
                <Header />
            </TestingRouter>
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('Should render unauthenticated header component', () => {
        const tree = renderer.create(
            <TestingRouter user={userUnauthDummy}>
                <Header />
            </TestingRouter>
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })
})