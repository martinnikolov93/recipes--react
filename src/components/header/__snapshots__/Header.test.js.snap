// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should render authenticated header component 1`] = `
<header>
  <h2
    className="app-title"
  >
    <img
      alt="logo"
      className="logo"
      src="/logo48.png"
    />
    React Recipes
  </h2>
  <nav>
    <div
      className="nav-buttons-wrapper"
    >
      <a
        aria-current="page"
        className="nav-button nav-button-active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
      <span>
        <a
          aria-current={null}
          className="nav-button"
          href="/categories"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          Categories
        </a>
      </span>
      <a
        aria-current={null}
        className="nav-button"
        href="/recipe/add"
        onClick={[Function]}
      >
        Add Recipe
      </a>
      <a
        aria-current={null}
        className="nav-button"
        href="/favourites"
        onClick={[Function]}
      >
        Favourites
      </a>
      <a
        aria-current={null}
        className="nav-button"
        href="/profile"
        onClick={[Function]}
      >
        Profile
      </a>
      <a
        aria-current={null}
        className="nav-button"
        href="/logout"
        onClick={[Function]}
      >
        Logout
      </a>
    </div>
  </nav>
</header>
`;

exports[`Header component Should render unauthenticated header component 1`] = `
<header>
  <h2
    className="app-title"
  >
    <img
      alt="logo"
      className="logo"
      src="/logo48.png"
    />
    React Recipes
  </h2>
  <nav>
    <div
      className="nav-buttons-wrapper"
    >
      <a
        aria-current="page"
        className="nav-button nav-button-active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
      <span>
        <a
          aria-current={null}
          className="nav-button"
          href="/categories"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          Categories
        </a>
      </span>
      <a
        aria-current={null}
        className="nav-button"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
      <a
        aria-current={null}
        className="nav-button"
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </div>
  </nav>
</header>
`;
